Dockerfile

```diff
@@ -1,42 +1,13 @@
-FROM quay.io/condaforge/mambaforge
+FROM python:3.10.8
+WORKDIR /home/pandas
 
-# if you forked pandas, you can pass in your own GitHub username to use your fork
-# i.e. gh_username=myname
-ARG gh_username=pandas-dev
-ARG pandas_home="/home/pandas"
+RUN apt-get update && apt-get -y upgrade
+RUN apt-get install -y build-essential
 
-# Avoid warnings by switching to noninteractive
-ENV DEBIAN_FRONTEND=noninteractive
+# hdf5 needed for pytables installation
+RUN apt-get install -y libhdf5-dev
 
-# Configure apt and install packages
-RUN apt-get update \
-    && apt-get -y install --no-install-recommends apt-utils git tzdata dialog 2>&1 \
-    #
-    # Configure timezone (fix for tests which try to read from "/etc/localtime")
-    && ln -fs /usr/share/zoneinfo/Etc/UTC /etc/localtime \
-    && dpkg-reconfigure -f noninteractive tzdata \
-    #
-    # cleanup
-    && apt-get autoremove -y \
-    && apt-get clean -y \
-    && rm -rf /var/lib/apt/lists/*
-
-# Switch back to dialog for any ad-hoc use of apt-get
-ENV DEBIAN_FRONTEND=dialog
-
-# Clone pandas repo
-RUN mkdir "$pandas_home" \
-    && git clone "https://github.com/$gh_username/pandas.git" "$pandas_home" \
-    && cd "$pandas_home" \
-    && git remote add upstream "https://github.com/pandas-dev/pandas.git" \
-    && git pull upstream main
-
-# Set up environment
-RUN mamba env create -f "$pandas_home/environment.yml"
-
-# Build C extensions and pandas
-SHELL ["mamba", "run", "--no-capture-output", "-n", "pandas-dev", "/bin/bash", "-c"]
-RUN cd "$pandas_home" \
-    && export \
-    && python setup.py build_ext -j 4 \
-    && python -m pip install --no-build-isolation -e .
+RUN python -m pip install --upgrade pip
+RUN python -m pip install --use-deprecated=legacy-resolver \
```

Comment:
`--use-deprecated=legacy-resolver` is in place because the new pip resolver seems to download multiple package versions to (I assume) check dependencies via setup.py or pyproject.yml. Some of our list dependencies like boto3 have a ton of releases, so it makes for a painfully slow process

Possible resolutions to use the newer solver I think are

  1. https://github.com/pandas-dev/pandas/issues/48828
  2. https://pip.pypa.io/en/stable/topics/dependency-resolution/#possible-ways-to-reduce-backtracking

So either reduce the number of dependencies, add a floor to their version or set up a constraint file once after solving

Reply:
Or use mamba, which has a decent dependency resolver ...